File: javascriptp/OOPS/js_cohesion/book.js
var Book = function () {
  this.add = function (num) {
    return num + 2;
  };

  this.substract = function (num) {
    return num - 2;
  };

  this.multiple = function (num) {
    return num * 2;
  };

  this.divide = function (num) {
    return num / 2;
  };

  this.weaklyCohesive = function (num) {
    return this.divide(this.multiple(this.substract(this.add(num))));
  };

  this.highlyCohesive = function (num) {
    return ((num + 2 - 2) * 2) / 2;
  };
};

var b1 = new Book();
console.log("highlyCohesive", b1.highlyCohesive(2));
console.log("weaklyCohesive", b1.weaklyCohesive(2));

File: pythonp/OOPS/py_cohesion/book.py
class Book:
    def add(self, num):
        return num + 2

    def substract(self, num):
        return num - 2

    def multiple(self, num):
        return num * 2

    def divide(self, num):
        return num / 2

    def weaklyCohesive(self, num):
        return self.divide(self.multiple(self.substract(self.add(num))))

    def highlyCohesive(self, num):
        return ((num + 2 - 2) * 2) / 2


b1 = Book()
print("highlyCohesive", b1.highlyCohesive(2))
print("weaklyCohesive", b1.weaklyCohesive(2))
